test/common/upstream/upstream_impl_test.cc

```diff
@@ -2813,6 +2813,49 @@ TEST_F(ClusterInfoImplTest, DefaultConnectTimeout) {
   EXPECT_EQ(std::chrono::seconds(5), cluster->info()->connectTimeout());
 }
 
+TEST_F(ClusterInfoImplTest, MaxConnectionDurationTest) {
+  const std::string yaml_default_max_connection_duration = R"EOF(
+  name: cluster1
+  type: STRICT_DNS
+  lb_policy: ROUND_ROBIN
+  )EOF";
+
+  const std::string yaml_set_max_connection_duration = R"EOF(
+  name: cluster2
+  type: STRICT_DNS
+  lb_policy: ROUND_ROBIN
+  typed_extension_protocol_options:
+    envoy.extensions.upstreams.http.v3.HttpProtocolOptions:
+      "@type": type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions
+      explicit_http_config:
+        http_protocol_options: {}
+      common_http_protocol_options:
+        max_connection_duration: 9s
+  )EOF";
+
+  const std::string yaml_zero_max_connection_duration = R"EOF(
+  name: cluster2
+  type: STRICT_DNS
+  lb_policy: ROUND_ROBIN
+  typed_extension_protocol_options:
+    envoy.extensions.upstreams.http.v3.HttpProtocolOptions:
+      "@type": type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions
+      explicit_http_config:
+        http_protocol_options: {}
+      common_http_protocol_options:
+        max_connection_duration: 0s
+  )EOF";
```

Comment:
are these two strings identical except the "0s" vs "9s"? You could use fmt::format to factor that out too :)

Reply:
Done!